### ISSUE TITLE
TODO: Look into using "Full-Stack-Fortran" Fortran/Emscripten toolchain 

### ISSUE THREAD

**Issue by hoodmane** [pyodide/pyodide#2318]
See https://github.com/StarGate01/Full-Stack-Fortran

**Comment by rth** [comment]
Yes, this one was one of the promising leads in https://github.com/pyodide/pyodide/issues/184#issuecomment-616948624

**Comment by rth** [comment]
No worries) Their setup is in https://github.com/StarGate01/Full-Stack-Fortran/blob/master/tools/f90wasm/Dockerfile and in particular, in involves [building a patched gfortran](https://github.com/StarGate01/Full-Stack-Fortran/blob/master/tools/f90wasm/Dockerfile#L30)  and  DragonEgg gcc plugin, which I assume must have exactly the same versions of gcc installed on the system. So that would make the build setup much more constrained, though maybe it's still worth it.

**Comment by hoodmane** [comment]
Yeah the big advantage of f2c is that it doesn't require hooking into the compiler toolchain. Big disadvantage is that it doesn't work all that well.

**Comment by agriyakhetarpal** [comment]
We should now be able to use Flang, which as of LLVM 19 can now compile to WASM directly. I also opened #5030 as a separate item. Our friends at emscripten-forge are trying it out for building SciPy: emscripten-forge/recipes#1288

**Comment by IsabelParedes** [comment]
> Our friends at emscripten-forge are trying it out for building SciPy: https://github.com/emscripten-forge/recipes/pull/1288

We've used flang for `blas`, `lapack`, and `r-base` also.
